cloudinary-core/src/main/java/com/cloudinary/Url.java

```diff
@@ -417,6 +418,11 @@ public String generate(String source) {
             } catch (MalformedURLException ignored) {
             }
         }
+        if (AnalyticsUtils.token != null) {
+            if(!AnalyticsUtils.checkIfQueryParamExist(url)) {
+                url = (new StringBuilder()).append(url).append(AnalyticsUtils.analyticsPrefix).append(AnalyticsUtils.token).toString();
```

Comment:
Why don't we use `URL` class(like with  `authToken` above) and perform less safe string manipulations?